general/device.hpp

```diff
@@ -15,6 +15,12 @@
 #include "globals.hpp"
 #include "mem_manager.hpp"
 
+
+#ifdef MFEM_USE_CEED
+// FES -> CeedBasis, CeedElemRestriction hash tables
+#include <unordered_map>
+#endif
```

Comment:
You don't use `unordered_map` in this file, this include can go directly in the cpp file.

Reply:
I put it here so that it's included in both `device.cpp` and `ceed.cpp`? Is including separately in those two files preferable?

Reply:
I might also need it to get the new function signatures in `device.cpp` and `ceed.cpp` for the tuple->hash function I'm going to re-implement here, right?

Reply:
In general it's better to include files in the `cpp` files, rather than the `hpp`, if you can. This prevents including unwanted libraries when including a `hpp` file, which can result in various forms of trouble.